R/coord-munch.r

```diff
@@ -9,11 +9,17 @@
 #'   All other variables are duplicated as needed.
 #' @param range Panel range specification.
 #' @param segment_length Target segment length
+#' @param make_closed Whether data should be converted to closed polygons before
+#'   munching.
 #' @keywords internal
 #' @export
-coord_munch <- function(coord, data, range, segment_length = 0.01) {
+coord_munch <- function(coord, data, range, segment_length = 0.01, make_closed = FALSE) {
   if (coord$is_linear()) return(coord$transform(data, range))
 
+  if (make_closed) {
+    data <- close_poly(data)
+  }
+
```

Comment:
if the shape is closed then the last coordinate (which would be a replicate of the first) should be removed before returning the new data

Reply:
I'm not sure I'm following along correctly. The aim of this PR is to add an option to add the last coordinate as a replicate of the first (i.e. close a polygon), not opening up polygons.

Reply:
Yes, we need to add the last coordinate so that the closing segment can be munched as well. But after the munching has been done we should remove the last coordinate again so that it is not replicated in the boundary

Reply:
This should happen lower down in the function of course (after the `munch_data()` call), so maybe my comment was misplaced